src/Streamly.hs

```diff
@@ -115,45 +117,55 @@ module Streamly
     , AsyncT
     , WAsyncT
     , ParallelT
+    , mkAsync
 
     -- ** Zipping Streams
     -- $zipping
     , ZipSerialM
     , ZipAsyncM
 
-    -- * Parallel Function Application
-    -- $application
-    , (IP.|$)
-    , (IP.|&)
-    , (IP.|$.)
-    , (IP.|&.)
-    , mkAsync
+    -- * IO Streams
+    , Serial
+    , WSerial
+    , Ahead
+    , Async
+    , WAsync
+    , Parallel
+    , ZipSerial
+    , ZipAsync
 
-    -- * Merging Streams
-    -- $sum
+    -- * Re-exports
+    , Semigroup (..)
+
+    -- * Deprecated
+    -- ** Parallel Function Application
+    , (|$)
+    , (|&)
+    , (|$.)
+    , (|&.)
+
+    -- ** Merging Streams
     , serial
     , wSerial
     , ahead
     , async
     , wAsync
     , parallel
 
-    -- * Concurrency Control
-    -- $concurrency
+    -- ** Concurrency Control
     , maxThreads
     , maxBuffer
 
-    -- * Rate Limiting
-    , Rate (..)
+    -- ** Rate Limiting
+    , Rate
```

Comment:
We are not exposing the constructors anymore? This can break existing code.

Reply:
No idea how to deprecate constructors

Reply:
Existing code will break even without this change.

Reply:
Yes, but least breakage is preferred. Not a big deal though.